app/model/repository/Comments.php

```diff
@@ -31,4 +32,57 @@ public function persistComment(Comment $comment) {
   public function persistThread(CommentThread $thread) {
     $this->persist($thread);
   }
+
+  /**
+   * Get the number of visible comments for given user
+   * @param $threadId Id of the comments thread.
+   * @param $userId Id of the viewer or null (if nobody is logged in).
+   * @param bool $allVisible True = all coments visible for the user, false = comments made by the user.
+   */
+  public function getThreadCommentsCount(CommentThread $thread, ?User $user, bool $allVisible = true)
```

Comment:
The meaning of the boolean flag is not obvious in the usages of this method. How about making it private and implementing two methods with different names that call this one?

Reply:
Can do.